src/wp-includes/functions.php

```diff
@@ -2079,8 +2080,30 @@ function wp_mkdir_p( $target ) {
 		$dir_perms = 0777;
 	}
 
+	/**
+	 * Fires before the directory creation
+	 *
+	 *
+	 * @since 6.5 The action was added.
+	 *
+	 * @param strin $target Full path to attempt to create
+	 * @param int $dir_perms Directory permissions
+	 */
+	do_action( 'before_create_directory', $target, $dir_perms );
+
 	if ( @mkdir( $target, $dir_perms, true ) ) {
 
+		/**
+		 * Fires after the directory is created and the permissions are set.
+		 *
+		 *
+		 * @since 6.5 The action was added.
+		 *
+		 * @param strin $target Full path to created directory
+		 * @param int $dir_perms Directory permissions
+		 */
```

Comment:
```suggestion
		/**
		 * Fires after the directory is created and the permissions are set.
		 *
		 * @since 6.5.0
		 *
		 * @param string $target    Full path to created directory.
		 * @param int    $dir_perms Directory permissions.
		 */
```